http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/utils/ChannelResolver.java

```diff
@@ -21,26 +21,37 @@
 import io.netty.channel.ChannelFactory;
 import io.netty.channel.EventLoopGroup;
 import io.netty.channel.ReflectiveChannelFactory;
+import io.netty.channel.epoll.EpollDatagramChannel;
 import io.netty.channel.epoll.EpollEventLoopGroup;
 import io.netty.channel.epoll.EpollSocketChannel;
 import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.DatagramChannel;
+import io.netty.channel.socket.nio.NioDatagramChannel;
 import io.netty.channel.socket.nio.NioSocketChannel;
 import java.util.HashMap;
 import java.util.Map;
 import software.amazon.awssdk.annotations.SdkInternalApi;
 import software.amazon.awssdk.http.nio.netty.internal.DelegatingEventLoopGroup;
 
 @SdkInternalApi
-public final class SocketChannelResolver {
+public final class ChannelResolver {
 
-    private static final Map<String, String> KNOWN_EL_GROUPS = new HashMap<>();
+    private static final Map<String, String> KNOWN_EL_GROUPS_CHANNELS = new HashMap<>();
+    private static final Map<String, String> KNOWN_EL_GROUPS_DATAGRAMS = new HashMap<>();
 
     static {
-        KNOWN_EL_GROUPS.put("io.netty.channel.kqueue.KQueueEventLoopGroup", "io.netty.channel.kqueue.KQueueSocketChannel");
-        KNOWN_EL_GROUPS.put("io.netty.channel.oio.OioEventLoopGroup", "io.netty.channel.socket.oio.OioSocketChannel");
+        KNOWN_EL_GROUPS_CHANNELS.put("io.netty.channel.kqueue.KQueueEventLoopGroup",
+                                     "io.netty.channel.kqueue.KQueueSocketChannel");
+        KNOWN_EL_GROUPS_CHANNELS.put("io.netty.channel.oio.OioEventLoopGroup",
+                                     "io.netty.channel.socket.oio.OioSocketChannel");
+
+        KNOWN_EL_GROUPS_DATAGRAMS.put("io.netty.channel.kqueue.KQueueEventLoopGroup",
+                                      "io.netty.channel.kqueue.KQueueDatagramChannel");
+        KNOWN_EL_GROUPS_DATAGRAMS.put("io.netty.channel.oio.OioEventLoopGroup",
+                                      "io.netty.channel.socket.oio.OioDatagramChannel");
```

Comment:
The styling was complaining for the long lines, so I decided to split them like this.